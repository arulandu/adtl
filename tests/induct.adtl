s : Type 0.
e1 := fun x : Nat => s.
e2 : e1 0.
e3 : (x : Nat) -> e1 x -> e1 (succ x).
e4 := succ 0.
e5 := elimNat e1 e2 e3 e4.
Context.