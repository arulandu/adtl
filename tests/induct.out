s : Type 0
e1 = fun _ : Nat => s
    : Nat -> Type 0
e2 : e1 0
e3 : forall x : Nat, e1 x -> e1 Succ x
e4 = Succ 0
    : Nat
e5 = ElimNat (e1) (e2) (e3) (e4)
    : s
